chain/chain/src/resharding/manager.rs

```diff
@@ -105,8 +105,25 @@ impl ReshardingManager {
             return Ok(());
         }
 
+        let tracked_children_shards = split_shard_event
+            .children_shards()
+            .iter()
+            .filter(|child_shard_uid| {
+                self.shard_tracker.cares_about_shard_this_or_next_epoch(
```

Comment:
sanity check: Is the resharding block passed as the `prev_block` to this method? We should be checking if the shard is tracked in the first or second epoch with new shard layout.

Reply:
I actually think this checks if the child is tracked in current epoch (of course not, it doesn't exist) or in the first epoch after resharding

So potentially, if the node tracks the child in the epoch after resharding, we are skipping resharding. 

This is the same logic for which we created the mapping before, and also the same logic used to decide if we do TrieResharding of a child

@shreyan-gupta can this be problematic? I'm a bit lost ;)

Reply:
Perhaps it is fine because `resharding_block_hash` is passed as `prev_block` hash to `cares_about_shard_in_epoch_from_prev_hash`.

Reply:
- So the `cares_about_shard_this_or_next_epoch` takes `prev_hash` as input.
- If we are passing the resharding block here, i.e. last block of the epoch with old shard layout
- technically we would be checking if the shard_uid is being tracked in the resharding epoch and the epoch after (the first and second epoch in which we have the new shard layout).
- Basically I agree with what wac said in the original comment and this should be fine.